### DISCUSSION TITLE
Problems with gnssdump，thanks for the answer.

### DISCUSSION THREAD

**Discussion by qnxsgwy** [semuconsulting/pygnssutils#64] [Q&A]
sorry, I’m a newbie to the world of python and programming.
How does the gnssdump utility output the filtered results to a serial port or socket? 
The example says:"The custom output handler can be a writeable output medium (serial, file, socket or queue) or an evaluable Python expression (e.g. lambda function)".
But there is only a specific command for outputting to a file. Please tell me what I have to do to filter the information coming from the serial port or socket and send it back. Thank you very much!

**Comment by semuadmin** [comment]
Command line options are limited - for more flexibility, use the [GNSSStreamer](https://www.semuconsulting.com/pygnssutils/pygnssutils.html#module-pygnssutils.gnssdump) class within a simple Python script.

Have a look at the [gnssstreamer_socket.py](https://github.com/semuconsulting/pygnssutils/blob/main/examples/gnssstreamer_socket.py) example.
